LSPS1/LSPS1.yaml

```diff
@@ -0,0 +1,430 @@
+      parameters:
+        - in: path
+          name: id
+          description: "Order id."
+          required: true
+          schema:
+              type: string
+      requestBody:
+          description: "Information needed to open the channel."
+          required: true
+          content:
+              application/json:
+                  schema:
+                      $ref: '#/components/schemas/ChannelOpenRequest'
+      responses:
+        "201":
+          description: "Success"
+          content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/ChannelOpenResponse"
+        "400":
+          description: "Validation error"
+        "404":
+          description: "Not found"
+        "412":
+          description: "Channel open failed"
+          content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/ChannelOpenResponse"
+
+
+components:
+    schemas:
+      ChannelApiInfo:
+        type: "object"
+        properties:
+          version:
+            type: integer
+            description: "Version of this api. See https://github.com/BitcoinAndLightningLayerSpecs/lsp"
+            example: 2
+          extensions:
+            type: array
+            items:
+              $ref: "#/components/schemas/Extension"
+          website:
+            type: string
+            description: "Website that provides devs a way to get in contact and/or TOS."
+            example: "https://blocktank.to/contact"
+
+      # ChannelApiInfo String Extension      
+      BaseExtension:
+        type: object
+        properties:
+          name:
+            type: string
+            example: 'local_balance'
+            description: Extension name.
+          version:
+            type: integer
+            example: 1
+            default: 1
+            nullable: true
+            description: Version of this extension. Integer starting at 1 counting up.
+        
+      OnchainPaymentExtension:
+        # Extends BaseExtension
+        allOf:
+          - $ref: "#/components/schemas/BaseExtension"
+          - type: object
+            properties:
+              min_satoshi:
+                type: string
+                description: "Minimal number of satoshi required to be eligable for an onchain payment."
+                example: 50000
+
+      Extension:
+      # One of these types need to match
+        oneOf:
+          - type: string
+          - $ref: "#/components/schemas/BaseExtension"
+          - $ref: "#/components/schemas/OnchainPaymentExtension"
+    
+      ChannelOrderRequest:
+        type: "object"
+        required:
+          - remote_balance
+        properties:
+          remote_balance:
+            type: integer
+            description: "Inbound liquidity amount in satoshis. Funds are at the LSP side."
+            example: 5000000
+            minimum: 0
+          local_balance:
```

Comment:
Thinking about this a little more. An option could be that every endpoint's requests and responses are constructed with envelope object, where extensions are able to add fields to. So `ChannelOrderRequest` could become:
```
{
    "version": 1 // spec version
    "channel_order_request": {
       // the actual request as defined by the base spec
    },
    "local_balance": { 
        "version": 12,
        // values for the local balance extension version 12 go here
    }
}
```

Reply:
### API Key
All things that are about api-keys, discounts or any custom modification of the order, I added the `coupon_code` for. Originally, I added this field for channel sales, in case you want to give the customer 10% off or so. Nothing speaks against using it with an apiKey IMO. 

### Request/Response structure
That's an interesting approach. I am a bit worried we over engineer the api though.